pkg/server/asyncworker.go

```diff
@@ -33,17 +36,18 @@ import (
 type AsyncWorker struct {
 	worker.Service
 
+	options        hostoptions.HostOptions
 	handlerBuilder []builder.Builder
 }
 
 // NewAsyncWorker creates new service instance to run AsyncRequestProcessWorker.
 func NewAsyncWorker(options hostoptions.HostOptions, builder []builder.Builder) *AsyncWorker {
 	return &AsyncWorker{
-		Service: worker.Service{
-			ProviderName: "radius",
-			Options:      options,
-		},
+		options:        options,
 		handlerBuilder: builder,
+		Service:        worker.Service{
+			// Will be initialized later
```

Comment:
The other one is initialized in the **Run** function but this one is in the **init**. Should they be consistent?

Reply:
This one (applications-rp) has a lot more logic because it's initializing a lot of dependencies. I don't feel like it's critical to make the implementation details of these structs consistent.